test/test_torrent.cpp

```diff
@@ -285,16 +299,16 @@ TORRENT_TEST(torrent)
 		file_storage fs;
 
 		fs.add_file("test_torrent_dir2/tmp1", 1024);
-		libtorrent::create_torrent t(fs, 128 * 1024, 6);
+		libtorrent::create_torrent t(fs, 1024, 6);
```

Comment:
this used to be 128 blocks

Reply:
The second argument is a `piece_size`. Setting it back to 128 * 1024 will require additional changes in `test_running_torrent()`.

Reply:
Alternative fix 13e92c31.

Reply:
A lot of this buffer juggling is a lot nicer in master, with `span`

Reply:
The problem is not in particular buffer type, it is about how piece size for the single-file torrent is being obtained.

Reply:
I don't think I actually understand this (yet). Is this an independent problem from setting the file priorities?

Reply:
From this test I discovered the problem for the first time. At that moment I didn't have a clue on what's causing test failure. So I began to investigate and had found some issues in it. When I fixed em', test still was failing randomly. Soon I found a root cause and it was file priorities, so I fixed it too.

TL;DR: Test issue appears to be independent, and could be submitted as a separate PR.

Reply:
Without that changes test should be passing, but will take extra 10 seconds to finish. Also it is error prone, because of incorrect piece handling logic.

Reply:
You decide, whether fixing test issues goes to this PR or another one. I just want this test be correct in the end.